src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -43,10 +44,15 @@ def post(self, request: Request):
         assert relay is not None  # should be provided during Authentication
         response = {}
 
-        full_config_requested = request.relay_request_data.get("fullConfig")
-
-        if full_config_requested and not relay.is_internal:
-            return Response("Relay unauthorized for full config information", 403)
+        if not request.relay_request_data.get("fullConfig"):
+            # We capture a message to make it apparent to self-hosted users that now it's only possible to query
+            # full project configs.
+            sentry_sdk.capture_message(
+                "Requesting the non-full configuration is not possible anymore"
+            )
+            return Response(
+                "Requesting the non-full configuration is not possible anymore", status=400
+            )
```

Comment:
We still need the `is_internal` check, but that also means we can simplify and ignore the parameter entirely IMO. The 403 should be signal enough.
```suggestion
        if not relay.is_internal:
            return Response("Relay unauthorized for config information", 403)
```

Reply:
Yeah I was wondering if we wanted that, I thought we wanted to allow external relays to get the config. My bad for misunderstanding